cmake/modules/extensions.cmake

```diff
@@ -2507,76 +2538,61 @@ function(zephyr_get variable)
+  # environment value or when appending with the MERGE operation.
+  foreach(var ${GET_VAR_VAR})
+    set(current_${var} ${${var}})
+    set(${var})
 
-  if(DEFINED sysbuild_${variable})
-    if(GET_VAR_MERGE)
-      list(APPEND ${variable} ${sysbuild_${variable}})
-    else()
-      set(${variable} ${sysbuild_${variable}} PARENT_SCOPE)
-      return()
-    endif()
-  endif()
-  if(SYSBUILD AND GET_VAR_MERGE AND NOT used_global AND "${GET_VAR_SYSBUILD}" STREQUAL "GLOBAL")
-    get_property(sysbuild_${variable} TARGET sysbuild_cache PROPERTY ${variable})
-    list(APPEND ${variable} ${sysbuild_${variable}})
-  endif()
-  if(DEFINED CACHE{${variable}})
-    if(GET_VAR_MERGE)
-      list(APPEND ${variable} $CACHE{${variable}})
-    else()
-      set(${variable} $CACHE{${variable}} PARENT_SCOPE)
-      return()
-    endif()
-  endif()
-  if(DEFINED ENV{${variable}})
-    if(GET_VAR_MERGE)
-      list(APPEND ${variable} $ENV{${variable}}})
+    if(SYSBUILD)
+      get_property(sysbuild_name TARGET sysbuild_cache PROPERTY SYSBUILD_NAME)
+      get_property(sysbuild_main_app TARGET sysbuild_cache PROPERTY SYSBUILD_MAIN_APP)
+      get_property(sysbuild_${var} TARGET sysbuild_cache PROPERTY ${sysbuild_name}_${var})
+      if(NOT DEFINED sysbuild_${var} AND
+         ("${GET_VAR_SYSBUILD}" STREQUAL "GLOBAL" OR sysbuild_main_app)
+      )
+        get_property(sysbuild_${var} TARGET sysbuild_cache PROPERTY ${var})
+      endif()
     else()
-      set(${variable} $ENV{${variable}} PARENT_SCOPE)
-      # Set the environment variable in CMake cache, so that a build invocation
-      # triggering a CMake rerun doesn't rely on the environment variable still
-      # being available / have identical value.
-      set(${variable} $ENV{${variable}} CACHE INTERNAL "")
+      set(sysbuild_${var})
     endif()
+  endforeach()
 
-    if(DEFINED ${variable} AND NOT "${${variable}}" STREQUAL "$ENV{${variable}}")
-      # Variable exists as a local scoped variable, defined in a CMakeLists.txt
-      # file, however it is also set in environment.
-      # This might be a surprise to the user, so warn about it.
-      message(WARNING "environment variable '${variable}' is hiding local "
-                      "variable of same name.\n"
-                      "Environment value (in use): $ENV{${variable}}\n"
-                      "Local scope value (hidden): ${${variable}}\n"
-      )
-    endif()
+  set(scopes "sysbuild;CACHE;ENV;current")
+  foreach(scope IN LISTS scopes)
+    foreach(var ${GET_VAR_VAR})
+      zephyr_var_name("${var}" "${scope}" expansion_var)
+      if(DEFINED expansion_var)
+        string(CONFIGURE "${expansion_var}" scope_value)
+        if(GET_VAR_MERGE)
+          list(APPEND ${variable} ${scope_value})
+        else()
+          set(${variable} ${scope_value} PARENT_SCOPE)
+
+          if("${scope}" STREQUAL "ENV" AND DEFINED current_${var}
+	     AND NOT "${current_${var}}" STREQUAL "$ENV{${var}}")
```

Comment:
Adjust alignment here

Reply:
Something is not right here and I'm not sure if it's the file or github, in the file diff view, the "AND NOT" line is way left, in the comment view, it is right.

Reply:
fixed